### PR TITLE
add a section for ivy-gardener in deep dive document

### PR DESCRIPTION
**Pull request by Daniel4078**

### CODE REVIEW

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Looks great, but can you mention as well that the same notes you wrote also apply to our scheduled linting bot which run every day at 8:00 UTC? Both actually share the same code, with `ivy-gardener` just runs on the changed files in a PR, the scheduled lint bot works on all files on `master`

**Review by KareemMAX — CHANGES_REQUESTED** [review]
One small nitpick, let me know what you think

**Review by KareemMAX — CHANGES_REQUESTED** [review]
That would be misleading, when `ivy-gardener` shows an error that it failed it will always be a permission issue.

If there are errors that need human intervention, `ivy-gardener` will pass, but the linting check will fail.

Can you please change that?

File: docs/overview/deep_dive/formatting.rst [hunk 1]
```diff
@@ -169,9 +169,22 @@ We have a GitHub action that runs:
...
+a new commit if there is any. 
+
+However, it is possible for the linters run in the ivy-gardener and the github action every day to face formatting 
+errors that need human intervention like typos and uninitialized arguments. In this case, errors will be thrown 
+by the linters and by the lint checks that runs later, while the fixes for other normal errors will still be applied 
+as the ivy-gardener is passing properly.
+
+On the other hand, ivy-gardener itself can fail if the bot handling it (ivy-branch) can not apply the changes 
+suggested by the linters, for example, when it does not have access to edit the target branch. In this case, you 
+should check the logs of lint checks to manually resolve the formatting errors by commiting the changes yourself.
```
Comment (KareemMAX) [inline docs/overview/deep_dive/formatting.rst:187]: Here, you can simply give the maintainers access (this is an option in GitHub UI)

File: docs/overview/deep_dive/formatting.rst [hunk 2]
```diff
@@ -169,9 +169,22 @@ We have a GitHub action that runs:
...
+is to reformat the files you changed in your PR directly on GitHub. This is useful in case if you didn't setup 
+pre-commit correctly or if you or one of our maintainers want to reformat your code remotely.
+
+Under the hood, when ``ivy-gardener`` is found in a comment, a ivy bot will trigger the same set of lint checks 
+as in the pre-commit process. Then the suggested changes produced in the checks will be applied automatically as
+a new commit if there is any. 
+
+However, it is possible for the linters run in the ivy-gardener and the github action every day to face formatting 
+errors that need human intervention like typos and uninitialized arguments. In this case, errors will be thrown 
+by the linters and by the lint checks that runs later, while the fixes for other normal errors will still be applied
```
Comment (KareemMAX) [inline docs/overview/deep_dive/formatting.rst:182]: They are all errors, I believe normal is a bit inaccurate

```suggestion
by the linters and by the lint checks that runs later, while the fixes for other errors will still be applied 
```

File: docs/overview/deep_dive/formatting.rst [hunk 3]
```diff
@@ -169,9 +169,22 @@ We have a GitHub action that runs:
...
-of our maintainers want to reformat your code without having to clone the repository.
+The first action is to ensure that the code in the whole codebase is always formatted correctly. The second action 
+is to reformat the files you changed in your PR directly on GitHub. This is useful in case if you didn't setup 
+pre-commit correctly or if you or one of our maintainers want to reformat your code remotely.
+
+Under the hood, when ``ivy-gardener`` is found in a comment, a ivy bot will trigger the same set of lint checks 
+as in the pre-commit process. Then the suggested changes produced in the checks will be applied automatically as
+a new commit if there is any. 
+
+However, it is possible for the linters run in the ivy-gardener and the github action every day to face formatting
```
Comment (KareemMAX) [inline docs/overview/deep_dive/formatting.rst:180]:
```suggestion
However, it is possible for the linters run in the ``ivy-gardener`` and the GitHub action every day to face formatting 
```

File: docs/overview/deep_dive/formatting.rst [hunk 4]
```diff
@@ -169,9 +169,22 @@ We have a GitHub action that runs:
...
+pre-commit correctly or if you or one of our maintainers want to reformat your code remotely.
+
+Under the hood, when ``ivy-gardener`` is found in a comment, a ivy bot will trigger the same set of lint checks 
+as in the pre-commit process. Then the suggested changes produced in the checks will be applied automatically as
+a new commit if there is any. 
+
+However, it is possible for the linters run in the ivy-gardener and the github action every day to face formatting 
+errors that need human intervention like typos and uninitialized arguments. In this case, errors will be thrown 
+by the linters and by the lint checks that runs later, while the fixes for other normal errors will still be applied 
+as the ivy-gardener is passing properly.
```
Comment (KareemMAX) [inline docs/overview/deep_dive/formatting.rst:183]:
```suggestion
as the ``ivy-gardener`` is passing properly.
```

File: docs/overview/deep_dive/formatting.rst [hunk 5]
```diff
@@ -169,9 +169,22 @@ We have a GitHub action that runs:
...
+
+On the other hand, ivy-gardener itself can fail if the bot handling it (ivy-branch) can not apply the changes
```
Comment (KareemMAX) [inline docs/overview/deep_dive/formatting.rst:185]:
```suggestion
On the other hand, ``ivy-gardener`` itself can fail if the bot handling it (ivy-branch) can not apply the changes 
```

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @Daniel4078,
Here are some nitpicks. I'll leave them to you if you think they are not necessary please let me know and ask for a review afterwards.

Thank you so much

**Comment by Daniel4078** [comment]
@KareemMAX Done, I suppose it is pretty much done right?

**Review by KareemMAX — APPROVED** [review]
LGTM! Merging now

**Pull request merged**